src/main/java/com/google/devtools/build/lib/analysis/configuredtargets/PackageGroupConfiguredTarget.java

```diff
@@ -97,22 +124,36 @@ public PackageGroupConfiguredTarget(TargetContext targetContext, PackageGroup pa
     return packageSpecifications;
   }
 
-  @Override
-  public <P extends TransitiveInfoProvider> P getProvider(Class<P> provider) {
-    if (provider == FileProvider.class) {
-      return provider.cast(NO_FILES); // can't fail
-    } else {
-      return super.getProvider(provider);
-    }
-  }
-
   @Override
   protected Info rawGetStarlarkProvider(Provider.Key providerKey) {
+    if (providerKey.equals(PROVIDER.getKey())) {
+      return this;
+    }
     return null;
   }
 
   @Override
   protected Object rawGetStarlarkProvider(String providerKey) {
     return null;
   }
+
+  @StarlarkMethod(
+      name = "isAvailableFor",
+      documented = false,
+      parameters = {
+        @Param(
+            name = "label",
+            allowedTypes = {@ParamType(type = Label.class)})
+      },
+      useStarlarkThread = true)
+  public boolean starlarkMatches(Label label, StarlarkThread starlarkThread) throws EvalException {
+    RepositoryName repository =
+        BazelModuleContext.of(ofInnermostEnclosingStarlarkFunction(starlarkThread))
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;starlarkThread&#39; (index: 1 | type: StarlarkThread). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;label&#39;. The same parameter type is validated here for example: src/main/java/com/google/devtools/build/lib/analysis/starlark/StarlarkAttrModule.java:129. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).